training/distributed_training/pytorch/model_parallel_v2/gpt-neox/smp-train-gpt-neox-fsdp-tp.ipynb

```diff
@@ -0,0 +1,1032 @@
+    "    )\n",
+    "\n",
+    "hyperparameters.update(model_params)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Specify Essential Parameters for a SageMaker Training Job\n",
+    "\n",
+    "Next, you use the `SageMaker Estimator class` to define a SageMaker Training Job,\n",
+    "passing values through the following parameters for training job name,\n",
+    "the number of EC2 instances, the instance type,\n",
+    "and the size of the volume attached to the instances.\n",
+    "\n",
+    "- `instance_count`\n",
+    "- `instance_type`\n",
+    "- `volume_size`\n",
+    "- `base_job_name`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Update the Type and Number of EC2 Instance to Use\n",
+    "\n",
+    "The instance type and the number of instances you specify to the `instance_type` and `instance_count` parameters,\n",
+    "respectively, determine the total number of GPUs (`world_size`).\n",
+    "\n",
+    "$$\\text{world_size = (the number of GPUs on a single instance)} \\times \\text{(the number of instances)}$$"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# @title Setup instances: type and count.\n",
+    "\n",
+    "instance_type = \"ml.p4d.24xlarge\"\n",
+    "\n",
+    "# You need >= 1 p4d for 7b model.\n",
+    "# You need >= 8 p4d for 65b model.\n",
+    "instance_count = 1\n",
+    "\n",
+    "# Set to the number of GPUs on that instance.\n",
+    "processes_per_host = 8"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Specify a Base Job Name"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# @title Job name.\n",
+    "\n",
+    "instance_type_str = instance_type.split(\".\")[1] + instance_type.split(\".\")[2][:3]\n",
+    "\n",
+    "base_job_name = f'smp-{model_size:02d}b-{instance_type_str}-hs{hybrid_shard_degree}-tp{tensor_parallel_degree}-ao{offload_activations}-bs{hyperparameters[\"train_batch_size\"]:02d}'\n",
+    "\n",
+    "print(f\"Base job name: `{base_job_name}`.\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if not use_fsx:\n",
+    "    # If you want to resume training, set checkpoint_s3_uri to the same path as a previous job.\n",
+    "    # Previous checkpoint to load must have same model config.\n",
+    "    checkpoint_bucket = f\"s3://sagemaker-{region}-{account}/\"\n",
+    "    checkpoint_s3_uri = (\n",
+    "        f\"{checkpoint_bucket}/experiments/smp-fsdp-tp-{model_type}-checkpoints/{base_job_name}/\"\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# @title Image URI.\n",
+    "\n",
+    "image = (\n",
```

Comment:
I think we should not specify image, instead we should only specify `framework_version=2.0.1` in the `Estimator` and allow our changes to Python SDK to handle appropriate parsing of image URI based on instance type.

Reply:
Yes, just tested that `py_version="py310"` and `framework_version="2.0.1"` cannot pull the correct image.

Reply:
I believe our SDK changes havent been merged yet. I'd tested locally and it had worked

Reply:
WIP for me now, testing with a SM notebook.

Reply:
Was able to test it successfully. Will update it.

Reply:
Fixed as suggested, PTAL.